urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/typesystem/BasicTypeAdapter.java

```diff
@@ -0,0 +1,398 @@
+
+	    try { return this.equals(t.getShort()); }
+	    catch( BasicTypeException e ) { return false; }
+
+	} else if( this.getType() == TYPE_INT ) {
+
+	    try { return this.equals(t.getInt()); }
+	    catch( BasicTypeException e ) { return false; }
+
+	} else if( this.getType() == TYPE_LONG ) {
+
+	    try { return this.equals(t.getLong()); }
+	    catch( BasicTypeException e ) { return false; }
+
+	} else if( this.getType() == TYPE_FLOAT ) {
+
+	    try { return this.equals(t.getFloat()); }
+	    catch( BasicTypeException e ) { return false; }
+
+	} else if( this.getType() == TYPE_DOUBLE ) {
+
+	    try { return this.equals(t.getDouble()); }
+	    catch( BasicTypeException e ) { return false; }
+
+	} else if( this.getType() == TYPE_CHAR ) {
+
+	    try { return this.equals(t.getChar()); }
+	    catch( BasicTypeException e ) { return false; }
+
+	} else if( this.getType() == TYPE_STRING ) {
+
+	    try { return this.equals(t.getString()); }
+	    catch( BasicTypeException e ) { return false; }
+
+	} else if( this.getType() == TYPE_UUID ) {
+
+	    try { return this.equals(t.getUUID()); }
+	    catch( BasicTypeException e ) { return false; }
+
+	} else if( this.getType() == TYPE_ARRAY ) {
+
+	    try { return this.equals((BasicArrayType)t); }
+	    catch( BasicTypeException e ) { return false; } 
+	    catch( ClassCastException e ) { 
+		return false;  // throw new RuntimeException( "Failed to compare two BasicArrayTypes: " + e.getMessage(), e );
+	    }
+
+	} else {
+	    // INVALID TYPE IDENTIFIER???
+	    return false;
+	}
+    }
+
+    public boolean equals( boolean b ) {
+	try { return ( this.getBoolean() == b ); }
+	catch( BasicTypeException e ) { return false; }
+    }
+
+    public boolean equals( byte b ) {
+	try { return ( this.getByte() == b ); }
+	catch( BasicTypeException e ) { return false; }
+    }
+
+    public boolean equals( short s ) {
+	try { return ( this.getShort() == s ); }
+	catch( BasicTypeException e ) { return false; }
+    }
+
+    public boolean equals( int i ) {
+	try { return ( this.getInt() == i ); }
+	catch( BasicTypeException e ) { return false; }
+    }
+
+    public boolean equals( long l ) {
+	try { return ( this.getLong() == l ); }
+	catch( BasicTypeException e ) { return false; }
+    }
+
+    public boolean equals( float f ) {
+	try { return ( this.getFloat() == f ); }
+	catch( BasicTypeException e ) { return false; }
+    }
+    
+    public boolean equals( double d ) {
+	try { return ( this.getDouble() == d ); }
+	catch( BasicTypeException e ) { return false; }
+    }
+
+    public boolean equals( char c ) {
+	try { return ( this.getChar() == c ); }
+	catch( BasicTypeException e ) { return false; }
+    }
+
+    public boolean equals( String s ) {
+	try { 
+	    String tmp = this.getString();
+	    if( tmp == null && s != null )
+		return false;
+	    else if( tmp != null && s == null )
+		return false;
+	    else if( tmp == null && s == null )
+		return true;
+	    else
+		return tmp.equals(s);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 359:364, 373:378, 387:392.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.